Content.Server/Fluids/EntitySystems/SpillableSystem.cs

```diff
@@ -188,40 +188,35 @@ public bool TryGetPuddle(TileRef tileRef, [NotNullWhen(true)] out PuddleComponen
         // Get normalized co-ordinate for spill location and spill it in the centre
         // TODO: Does SnapGrid or something else already do this?
         var spillGridCoords = mapGrid.GridTileToWorld(tileRef.GridIndices);
-
-        var spillEntities = _entityLookup.GetEntitiesIntersecting(mapGrid.ParentMapId, spillGridCoords.Position).ToArray();
-        foreach (var spillEntity in spillEntities)
-        {
-            if (_solutionContainerSystem.TryGetRefillableSolution(spillEntity, out var solutionContainerComponent))
-            {
-                _solutionContainerSystem.Refill(spillEntity, solutionContainerComponent,
-                    solution.SplitSolution(FixedPoint2.Min(
-                        solutionContainerComponent.AvailableVolume,
-                        solutionContainerComponent.MaxSpillRefill))
-                );
-            }
-        }
+        var startEntity = EntityUid.Invalid;
+        PuddleComponent? puddleComponent = null;
 
         if (combine)
         {
+            var spillEntities = _entityLookup.GetEntitiesIntersecting(mapGrid.ParentMapId, spillGridCoords.Position)
+                .ToArray();
```

Comment:
Pretty sure there's a new method on IEntityLookup that lets you get a particular tile from a grid that's faster.